code/game/machinery/slotmachine.dm

```diff
@@ -138,49 +150,54 @@
 	balloon_alert(user, "machine rigged")
 	return TRUE
 
-/obj/machinery/computer/slot_machine/ui_interact(mob/living/user)
+/obj/machinery/computer/slot_machine/ui_interact(mob/living/user, datum/tgui/ui)
 	. = ..()
-	var/reeltext = {"<center><font face=\"courier new\">
-	/*****^*****^*****^*****^*****\\<BR>
-	| \[[reels[1][1]]\] | \[[reels[2][1]]\] | \[[reels[3][1]]\] | \[[reels[4][1]]\] | \[[reels[5][1]]\] |<BR>
-	| \[[reels[1][2]]\] | \[[reels[2][2]]\] | \[[reels[3][2]]\] | \[[reels[4][2]]\] | \[[reels[5][2]]\] |<BR>
-	| \[[reels[1][3]]\] | \[[reels[2][3]]\] | \[[reels[3][3]]\] | \[[reels[4][3]]\] | \[[reels[5][3]]\] |<BR>
-	\\*****v*****v*****v*****v*****/<BR>
-	</center></font>"}
-
-	var/dat
-	if(working)
-		dat = reeltext
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "SlotMachine", name)
+		ui.open()
 
-	else
-		dat = {"Five credits to play!<BR>
-		<B>Prize Money Available:</B> [money] (jackpot payout is ALWAYS 100%!)<BR>
-		<B>Credit Remaining:</B> [balance]<BR>
-		[plays] players have tried their luck today, and [jackpots] have won a jackpot!<BR>
-		<HR><BR>
-		<A href='?src=[REF(src)];spin=1'>Play!</A><BR>
-		<BR>
-		[reeltext]
-		<BR>"}
-		if(balance > 0)
-			dat+="<font size='1'><A href='?src=[REF(src)];refund=1'>Refund balance</A><BR>"
-
-	var/datum/browser/popup = new(user, "slotmachine", "Slot Machine")
-	popup.set_content(dat)
-	popup.open()
-
-/obj/machinery/computer/slot_machine/Topic(href, href_list)
-	. = ..() //Sanity checks.
+/obj/machinery/computer/slot_machine/ui_static_data(mob/user)
+	. = ..()
+	var/list/data = list()
+	data["icons"] = list()
+	for(var/icon_name in icons)
+		var/list/icon = icons[icon_name]
+		icon += list("icon" = icon_name)
+		data["icons"] += list(icon)
+	data["cost"] = SPIN_PRICE
+	data["jackpot"] = JACKPOT
+
+	return data
+
+/obj/machinery/computer/slot_machine/ui_data(mob/user)
+	. = ..()
+	var/list/data = list()
+	var/list/reel_states = list()
+	for(var/reel_state in reels)
+		reel_states += list(reel_state)
+	data["state"] = reel_states
+	data["balance"] = balance
+	data["working"] = working
+	data["money"] = money
+	data["plays"] = plays
+	data["jackpots"] = jackpots
+	data["paymode"] = paymode
+	return data
+
+
+/obj/machinery/computer/slot_machine/ui_act(action, list/params)
+	. = ..()
 	if(.)
-		return .
-
-	if(href_list["spin"])
-		spin(usr)
+		return
 
-	else if(href_list["refund"])
-		if(balance > 0)
-			give_payout(balance)
-			balance = 0
+	switch(action)
+		if("spin")
+			spin(usr)
```

Comment:
```suggestion
			spin(usr)
			return TRUE
```

returning TRUE will cause the UI to update with new values after you perform this action